jsonassert-assertj-java7/build.gradle

```diff
@@ -1,4 +1,5 @@
 dependencies {
-	compile project(':jsonassert')
-	compile 'org.assertj:assertj-core:2.3.0'
+	implementation project(':jsonassert')
+	implementation "org.assertj:assertj-core:${assertj7Version}"
```

Comment:
Does this result in the same lib in the end? Won't the `assertj-core` lib not be present (aka `implementation` ~ `compileOnly`)?

Reply:
I'm actually thinking of rewriting all of this to maven cause, let's put it nicely, I don't find pleasure in working with Gradle.

My understanding is that if you're using compile then you're not "doing things properly", but maybe I'm wrong.

Reply:
Yeh, I spent much more time using maven than Gradle so its still a learning curve for me and I have mixed feelings as well. :)

So `compile` -> `api` and `compileOnly` -> `implementation`. I think the equivalent of what it was before would be 
```java
dependencies {
	api project(':jsonassert')
	api "org.assertj:assertj-core:${assertj7Version}"
}
```

Reply:
@marcingrzejszczak, just stumbled across this.

It's not to say the compile is "not doing things properly", but more that it has inadvertent side effects to the compilation classpath of downstream consumers. That's a lot of the reason why Gradle separated the `compile` configuration into it's more detailed parts.

The premise is that `implementation` is available for compilation in Gradle, but becomes Maven `runtime` when being comsumed. This forces the consumer to declare that they need it for their own compilation if needed. `api` becomes `compile` like normal as it is then transitively available for compilation. 

This gets even more intricate when you start considering the javac AST of the classes that are used on the compilation classpath, which by using the `api` and `implementation` configurations enables Gradle to hide more detail from the project while also being able to further on it's goal for avoiding performing work whenever possible.